gridstatus/decorators.py

```diff
@@ -142,6 +142,13 @@ def wrapped_f(*args, **kwargs):
                 elif frequency == "MONTH_START":
                     frequency = pd.offsets.MonthBegin(1)
 
+                    # Since the date_range_maker is exclusive of the end date,
+                    # in the case where the end date is the first of the month,
+                    # we need to add a minute to the end date so that the
+                    # first of the month is included in the range.
+                    if args_dict["end"].day == 1:
```

Comment:
the mode explicitly adds the end date back in later on. are you sure this is necessary? 

https://github.com/kmax12/gridstatus/blob/696b4d206b81ae582b16fdd8455c5beb18ea3811/gridstatus/decorators.py#L161

Reply:
I'm not entirely sure. The failing test is `test_range_two_days_across_month` https://github.com/kmax12/gridstatus/blob/73a8419af6e4caae4ff5b531e5f685dff0c26388/gridstatus/tests/test_nyiso.py#L32 and it's failing [because the data only spans a single day](https://github.com/kmax12/gridstatus/actions/runs/7006327797/job/19057947311#step:6:3831). 

I think the problem is you need to retrieve two months of data, but the current method retrieves only one month (that of the `start_date`).

Reply:
i see. i'll have to take a look tomorrow. thanks for point to the test case

Reply:
I changed the fix, hopefully, it makes more sense. I also added another test in `test_nyiso` to be sure it's doing what we expect. `nyiso` was the only place where the codebase uses the decorator with "MONTH_START".

Reply:
okay, i took a look and i think the actual problem is the MonthBegin offset isn't working 

for example, this test case makes a call like this 

``` python
date_range_maker(
    start=pd.Timestamp('2023-11-01 05:00:00-0400', tz='US/Eastern'), 
    end=pd.Timestamp('2023-12-01 05:00:00-0500', tz='US/Eastern'), 
    freq=pd.offsets.MonthBegin(1)
    inclusive="neither"
)
```
which currently returns
```
[]
```

however, i would expect it to return the following is is the month start between those two dates

```
[pd.Timestamp('2023-12-01 00:00:00-0500', tz='US/Eastern')]
```

i think the solution is to make a custom offset like we do for [DayBeginOffset](https://github.com/kmax12/gridstatus/blob/7afdf1b588a0dfcbf38827c9b84e2f0bd7c4c156/gridstatus/decorators.py#L384) 

```python
class MonthBeginOffset:
    def __ladd__(self, other):
        return other.normalize() + pd.offsets.MonthBegin(1)

    def __radd__(self, other):
        return self.__ladd__(other)
```

if we swap in this offset, no other changes are needed and the test passes.

Reply:
this broke the `test_month_start_multiple_months` test. we'll can update that text to make it pass. i see multiple ways to do it. the important thing is that makes requests across multiple month